src/library_pthread.js

```diff
@@ -432,6 +432,13 @@ var LibraryPThread = {
           worker.onerror = function(e) {
             err('pthread sent an error! ' + e.filename + ':' + e.lineno + ': ' + e.message);
           };
+
+          if (ENVIRONMENT_HAS_NODE) {
+            worker.ref(); // TODO: investigate when we need this
```

Comment:
Ditto as below – I don’t think you need this while you have a `'message'` listener attached?